uspace/drv/hid/usbhid/usbhid.c

```diff
@@ -324,17 +325,89 @@ static errno_t usb_hid_init_report(usb_hid_dev_t *hid_dev)
 	return EOK;
 }
 
+static bool usb_hid_polling_callback(usb_device_t *dev, uint8_t *buffer,
+    size_t buffer_size, void *arg)
+{
+	if (dev == NULL || arg == NULL || buffer == NULL) {
+		usb_log_error("Missing arguments to polling callback.");
+		return false;
+	}
+	usb_hid_dev_t *hid_dev = arg;
+
+	assert(hid_dev->input_report != NULL);
+
+	usb_log_debug("New data [%zu/%zu]: %s", buffer_size,
+	    hid_dev->max_input_report_size,
+	    usb_debug_str_buffer(buffer, buffer_size, 0));
+
+	if (hid_dev->max_input_report_size >= buffer_size) {
+		/*! @todo This should probably be atomic. */
```

Comment:
We use `// TODO: `

Reply:
I'm not even sure this is our code, but sure we'll change it :)